linux/Platform.c

```diff
@@ -677,6 +691,106 @@ bool Platform_getNetworkIO(NetworkIOData* data) {
    return true;
 }
 
+void Platform_getBacktrace(Vector* frames, const Process* process, char** error) {
+#ifdef HAVE_LIBUNWIND_PTRACE
+   unw_addr_space_t addrSpace = unw_create_addr_space(&_UPT_accessors, 0);
+   if (!addrSpace) {
+      xAsprintf(error, "Unable to initialize libunwind.");
+      return;
+   }
+
+   const pid_t pid = Process_getPid(process);
+
+   if (pid <= 0) {
+      xAsprintf(error, "Unable to get the pid");
+      goto addr_space_error;
+   }
+
+   if (ptrace(PTRACE_ATTACH, pid, 0, 0) == -1) {
+      int ptraceErrno = errno;
+      xAsprintf(error, "ptrace: %s (%d)", strerror(ptraceErrno), ptraceErrno);
+      goto addr_space_error;
+   }
+
+   int waitStatus = 0;
+   if (wait(&waitStatus) == -1) {
+      int waitErrno = errno;
+      xAsprintf(error, "wait: %s (%d)", strerror(waitErrno), waitErrno);
+      goto ptrace_error;
+   }
+
+   if (WIFSTOPPED(waitStatus) == 0) {
+      *error = xStrdup("The process chosen is not stopped correctly.");
+      goto ptrace_error;
+   }
+
+   struct UPT_info* context = _UPT_create(pid);
+   if (!context) {
+      xAsprintf(error, "Unable to create the context of libunwind-ptrace");
+      goto ptrace_error;
+   }
+
+   unw_cursor_t cursor;
+   int ret = unw_init_remote(&cursor, addrSpace, context);
+   if (ret < 0) {
+      xAsprintf(error, "libunwind cursor: ret=%d", ret);
+      goto context_error;
+   }
+
+   int index = 0;
+   do {
+      char procName[2048] = "?";
+      unw_word_t offset;
+      unw_word_t pc;
+
+      if (unw_get_proc_name(&cursor, procName, sizeof(procName), &offset) == 0) {
+         ret = unw_get_reg(&cursor, UNW_REG_IP, &pc);
+         if (ret < 0) {
+            xAsprintf(error, "unable to get register rip : %d", ret);
+            break;
+         }
+
+         BacktraceFrame* frame = BacktraceFrame_new();
+         frame->index = index;
+         frame->address = pc;
+         frame->offset = offset;
+         frame->isSignalFrame = unw_is_signal_frame(&cursor);
+#if defined(HAVE_LIBIBERTY)
+         char* demangledName = cplus_demangle(procName,
+                                              DMGL_PARAMS | DMGL_ANSI | DMGL_VERBOSE | DMGL_RET_POSTFIX);
```

Comment:
Please use

```suggestion
         char* demangledName = cplus_demangle(procName, DMGL_PARAMS | DMGL_ANSI | DMGL_VERBOSE | DMGL_RET_POSTFIX);
```

or

```suggestion
         char* demangledName = cplus_demangle(procName,
            DMGL_PARAMS | DMGL_ANSI | DMGL_VERBOSE | DMGL_RET_POSTFIX);
```

Hanging indentation that skips half the horizontal space just to align with some opening parens is ridiculous …